### PR TITLE
Added a new dungeon inspired by Saw

### PR DESCRIPTION
**Pull request by mdj117**

### CODE REVIEW

**Comment by rzats** [comment]
LGTM! @flo you might want to take a look at this 🙂

File: assets/prefabs/structures/Saw.prefab [hunk 1]
```diff
@@ -0,0 +1,809 @@
+{
+  "StructureTemplate" : {
+    "type": "GooeysQuests:Saw",
```
Comment (flo) [inline assets/prefabs/structures/Saw.prefab]: There is no Saw type yet, and no structure spawns structures of that type. 

Have a look at the prefabs in the folder `assets/prefabs/structures/types`.

**Comment by flo** [comment]
Also there is a syntax error:

Unterminated array at line 784 column 25 path $.SpawnBlockRegions.regionsToFill[776]

**Comment by mdj117** [comment]
Will fix the syntax error, but should I define a new type or just define it as like a corridor ?
I think I'll define a new type :-)

**Comment by flo** [comment]
Adding a new structure type doesn't make sense when it does not get used.
Basically the structure type defines where it spawns. If you use a structure type that does not get spawned yet then your structure won't be spawned and no one will see it.

Please also have a look at my comments in the other PR. I also explain some basics there.

**Comment by flo** [comment]
For example the dungeonCorridorLeftTurnSpawner.prefab has the following section that makes it spawn a random structure template of type dungeon3x3StoneCorridor at the relative location of 1 0 3 facing left:
```
    "ScheduleStructurePlacement" : {
        "placementsToSchedule": [
            {
                "structureTemplateType": "GooeysQuests:dungeon3x3StoneCorridor",
                "front": "LEFT",
                "position": [1, 0, 3]
            }
        ]
    },
```

**Comment by flo** [comment]
If you made a new room that has a door sized entrance then use the type dungeon1x2StoneDoorFrame type. If it has a 3x3 wide entrance use dungeon3x3StoneCorridor. If it is a building for outside and a dungeon use type dungeonEntrance. You possibly then also want it to actually spawn a dungeon corridor (like with the json above) so that there is something bigger to explore.

**Comment by flo** [comment]
Do you still plan to add a spawn condition?

**Comment by mdj117** [comment]
as in ?

**Comment by flo** [comment]
Like
```
     "CheckBlockRegionCondition" : {
        "checksToPerform": [
            {"condition": "StructureTemplates:IsGroundLike", "region" :{"min": [-3, -1, 1], "size": [7, 7, 6]}}
        ]
    },
```
And have a look at existing entrance prefab for how to check for air.

**Comment by mdj117** [comment]
should I ?

**Comment by flo** [comment]
otherwise your structure will spawn at very bad locations. Like fore example over something the player built.

**Comment by mdj117** [comment]
Umm, Can I do that after GCI ?

**Comment by flo** [comment]
If this is your task then https://codein.withgoogle.com/dashboard/task-instances/4741269605580800/ I think you started again the wrong task.

**Comment by flo** [comment]
I tried it out, it still contains the old syntax error.

Once the syntax error is gone, you should be able to find it in the toolbox which can be obtained via "give toolbox" if you use the newest version of the StructureTemplates module.

File: assets/prefabs/structures/Saw.prefab [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+            { "blockType": "core:stone", "region": { "min": [24, -15, 3], "size": [7, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [8, -15, 4], "size": [14, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [8, -14, 4], "size": [23, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [9, -13, 4], "size": [22, 1, 2]}},
+            { "blockType": "core:stone", "region": { "min": [10, -12, 4], "size": [21, 2, 1]}},
+            { "blockType": "core:stone", "region": { "min": [11, -10, 4], "size": [3, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [11, -9, 4], "size": [14, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [12, -8, 4], "size": [13, 1, 1]}},
+            { "blockType": "core:stone", "region": { "min": [12, -7, 4], "size": [19, 1, 1]}},
+
```
Comment (flo) [inline assets/prefabs/structures/Saw.prefab]: Before "AddItemsToChest" you need to close the other array and component first properly.

**Comment by mdj117** [comment]
I'll do that in a bit
What do I have to do for the task I've currently claimed ?

**Comment by flo** [comment]
I don't know what your google account is. I suggest you just look up the task description

**Pull request closed without merging**